jobserver/views/workspaces.py

```diff
@@ -41,6 +43,60 @@ def post(self, request, *args, **kwargs):
         return redirect("/")
 
 
+@method_decorator(login_required, name="dispatch")
+class WorkspaceBackendFiles(View):
+    """
+    Orchestrate viewing of a Backend's "live" files for a given Workspace
+
+    We consume two URLs with one view, because we want to both do permissions
+    checks on the Workspace but also load the SPA for any given path under the
+    Workspace.
+    """
+
+    def get(self, request, *args, **kwargs):
+        workspace = get_object_or_404(
+            Workspace,
+            project__org__slug=self.kwargs["org_slug"],
+            project__slug=self.kwargs["project_slug"],
+            name=self.kwargs["workspace_slug"],
+        )
+
+        backend = get_object_or_404(Backend, name=self.kwargs["backend_slug"])
+
+        # ensure the user has access to the given Backend
+        if request.user not in backend.members.all():
+            raise Http404
+
+        # TODO: handle permissions here, I think we need a new permission for
+        # this?
+
+        f = furl(backend.level_4_url)
+
+        # build the base URL for which we want the auth token to authenticate,
+        # all paths beneath this one will be valid with the token
+        f.path.segments += ["workspace", workspace.name]
+        auth_token = build_auth_token(
+            url=request.build_absolute_uri(f.url),
```

Comment:
huh? Shouldn't this be building an absolute_uri for release-hatch, not for job-server?

Maybe I don't understand furl/django well enough here, the furl url is already absolute, no? What's the reasone for using build_absolute_uri from *this request*?

Anyway, see my comments above regards build_auth_token, I think the furl stuff could probably move in there, and we passing in `backend` and `workspace`?

Reply:
Hmm I now release you need the url as to pass to SPA.

Maybe return a tuple?

`return f.url, token.sign(backend.backend_token, salt="hatch")`

Reply:
I think I got confused over what URL this was being built for when writing an early version of this.  Coming back to it I'm a little confused too!